### PR TITLE
Update microsoft.dot net.helix.sdk

### PR DESCRIPTION
**Pull request by wli3**

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -7,6 +7,6 @@
   },
   "msbuild-sdks": {
     "Microsoft.DotNet.Arcade.Sdk": "1.0.0-beta.19474.3",
-    "Microsoft.DotNet.Helix.Sdk": "2.0.0-beta.19215.12"
+    "Microsoft.DotNet.Helix.Sdk": "5.0.0-beta.19524.2"
```
Comment (nguerrera) [inline global.json]: Are we supposed to use 5.0 version in 3.0 branch? Arcade is pushing out fixes from a servicing branch to .NET Core 3.x branches/channels.

Comment (wli3) [reply global.json]: @MattGal does Helix.sdk version number associated with dotnet core release? Should we just use the latest or the matching release version?

Comment (nguerrera) [reply global.json]:
```suggestion
    "Microsoft.DotNet.Helix.Sdk": "2.0.0-beta.19474.3"
```

This is from the same arcade build that we already take. Cop

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -19,5 +19,9 @@
       <Uri>https://github.com/NuGet/NuGet.Client</Uri>
       <Sha>c1f6915918b82c096bbd666bd9c18528f1f70630</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.DotNet.Helix.Sdk" Version="5.0.0-beta.19524.2">
```
Comment (nguerrera) [inline eng/Version.Details.xml]:
```suggestion
    <Dependency Name="Microsoft.DotNet.Helix.Sdk" Version="2.0.0-beta.19474.3">
```

File: eng/Version.Details.xml [hunk 3]
```diff
@@ -19,5 +19,9 @@
...
+      <Uri>https://github.com/dotnet/arcade</Uri>
+      <Sha />
```
Comment (nguerrera) [inline eng/Version.Details.xml]:
```suggestion
      <Sha>0e9ffd6464aff37aef2dc41dc2162d258f266e32</Sha>
```

Comment (nguerrera) [reply eng/Version.Details.xml]: Same as Arcade SDK.

Comment (nguerrera) [reply eng/Version.Details.xml]: Can you move this up next to Arcade and SignTool also.

Comment (nguerrera) [reply eng/Version.Details.xml]: HostModel and NuGet should be product, not toolset dependencies, but that`s for another change.

**Review by nguerrera — CHANGES_REQUESTED** [review]
Match the helix build to the arcade build we have.

**Comment by wli3** [comment]
@nguerrera just moved the order as well. it should be good now

**Review by nguerrera — APPROVED** [review]

**Comment by nguerrera** [comment]
Please squash

**Pull request merged**